rerun_py/rerun_sdk/rerun/_rerun2/log.py

```diff
@@ -147,8 +147,14 @@ def log(
             instanced[comp.extension_name] = comp.storage
         elif len(comp) == 1 and archetype_length > 1:
             splats[comp.extension_name] = comp.storage
-        elif len(comp) > 1:
+        elif len(comp) >= 1:
             instanced[comp.extension_name] = comp.storage
+        # TODO(#2825): For now we just don't log anything for unspecified components, to match the
+        # historical behavior.
+        # From the PoV of the high-level API, this is incorrect though: logging an archetype should
+        # give the user the guarantee that past state cannot leak into their data.
+        # else: # len == 0
+        #     instanced[comp.extension_name] = comp.storage
```

Comment:
advocating to just go ahead and make this change. `log` != `log_old_thing`
Guess it makes it easier though to implement `log_old_thing` with `log` :/

Reply:
Exactly, we need to fully kill the old API first, which means being done with the whole "implement the old in terms of the new" strategy!